### PR TITLE
Add a maven extension to for transforming Jakarta EE 8 based dependen…

### PR DESCRIPTION
**Pull request by jamezp**
…cies to Jakarta EE 9 based dependencies.

Note we may need to change this in the future to be less aggressive with transformation if we can find a good way to avoid it. However, this currently works and can be used for small transformations.

### CODE REVIEW

File: tools/eclipse-transformer-maven-extension/READEME.adoc [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+│   │       └── smallrye-common-expression-1.6.0-$$jakarta9$$.pom
+│   ├── smallrye-common-function
+│   │   ├── 1.6.0
+│   │   │   ├── _remote.repositories
+│   │   │   ├── smallrye-common-function-1.6.0.jar
+│   │   │   ├── smallrye-common-function-1.6.0.jar.sha1
+│   │   │   ├── smallrye-common-function-1.6.0.pom
+│   │   │   └── smallrye-common-function-1.6.0.pom.sha1
+│   │   └── 1.6.0-$$jakarta9$$-$$jakarta9$$
+│   │       ├── smallrye-common-function-1.6.0-$$jakarta9$$-$$jakarta9$$.jar
```
Comment (scottmarlow) [inline tools/eclipse-transformer-maven-extension/READEME.adoc:132]: Is the doubling of `-$$jakarta9$$` a typo or does that happen?

Comment (jamezp) [reply tools/eclipse-transformer-maven-extension/READEME.adoc:132]: I haven't been able to stop that from happening. It has to do with transitive dependencies I think, but I'm not sure where maven is pulling it from. Sometime I might have time to look again :)

Really a better solution will be to check the dependencies before this is invoked. However that is a lot more complicated.

Comment (scottmarlow) [reply tools/eclipse-transformer-maven-extension/READEME.adoc:132]: Will we see an extra `$$jakarta9$$` per transitive dependency level?  For example, do the transitive dependencies for smallrye-common have `$$jakarta9$$-$$jakarta9$$-$$jakarta9$$`?

Comment (jamezp) [reply tools/eclipse-transformer-maven-extension/READEME.adoc:132]: Not that I've seen. There seems to only be cases of a double suffix for some reason. It's very strange.

FWIW we may not end up needing this in RESTEasy if Infinispan releases a namespace change. We're moving Spring and MicroProfile stuff out of RESTEasy into their own projects. I can convert this to a draft if you'd prefer to ensure it doesn't get merged unless it's needed or until I can fix the double suffix.

Comment (scottmarlow) [reply tools/eclipse-transformer-maven-extension/READEME.adoc:132]: Thanks, converting to draft sounds like a good approach for now.  If we later decide that we need this and that users can work around the double suffix thing for transitive dependencies, we could follow that approach.

**Pull request closed without merging**